pkg/cmd/roachtest/tests/import_cancellation.go

```diff
@@ -0,0 +1,307 @@
+	// filename, `<tablename>.tbl.1`. Tables with a single file do not.
+	if numFiles > 1 {
+		return fmt.Sprintf(`'gs://cockroach-fixtures/tpch-csv/sf-100/%[1]s.tbl.%[2]d?AUTH=implicit'`, tableName, number)
+	}
+	return fmt.Sprintf(`'gs://cockroach-fixtures/tpch-csv/sf-100/%[1]s.tbl?AUTH=implicit'`, tableName)
+}
+
+func (t *importCancellationTest) runImportSequence(
+	ctx context.Context, rng *rand.Rand, tableName string, numFiles int,
+) {
+	n := t.c.All()[rng.Intn(len(t.c.All()))]
+	conn := t.c.Conn(ctx, t.L(), n)
+	if _, err := conn.Exec(`USE csv;`); err != nil {
+		t.Fatal(err)
+	}
+
+	filesToImport := make([]string, numFiles)
+	for i := 0; i < numFiles; i++ {
+		filesToImport[i] = t.makeFilename(tableName, i+1, numFiles)
+	}
+
+	attempts := randutil.RandIntInRange(rng, 2, 5)
+	for i := 0; i < attempts && len(filesToImport) > 0; i++ {
+		t.WorkerStatus(fmt.Sprintf(`attempt %d/%d for table %q`, i+1, attempts, tableName))
+		files := filesToImport
+		// If not the last attempt, import a subset of the files available.
+		// This will create MVCC range tombstones across separate regions of
+		// the table's keyspace.
+		if i != attempts-1 {
+			rng.Shuffle(len(files), func(i, j int) {
+				files[i], files[j] = files[j], files[i]
+			})
+			files = files[:randutil.RandIntInRange(rng, 1, len(files)+1)]
+		}
+		t.L().PrintfCtx(ctx, "Beginning import job for attempt %d/%d for table %s with %d files: %s",
+			i+1, attempts, tableName, len(files), strings.Join(files, ", "))
+
+		var jobID string
+		importStmt := fmt.Sprintf(`
+			IMPORT INTO %[1]s
+			CSV DATA (
+			%[2]s
+			) WITH  delimiter='|', detached
+		`, tableName, strings.Join(files, ", "))
+		err := conn.QueryRowContext(ctx, importStmt).Scan(&jobID)
+		if err != nil && !errors.Is(err, context.DeadlineExceeded) {
+			t.Fatal(err)
+		}
+
+		// If not the last attempt, cancel the job after a random duration in [10s,3m).
+		if i != attempts-1 {
+			timeout := time.Duration(randutil.RandIntInRange(rng, 10, 60*3)) * time.Second
+			select {
+			case <-time.After(timeout):
+				// fallthrough
+			case <-ctx.Done():
+				return
+			}
+
+			t.L().PrintfCtx(ctx, "Cancelling import job for attempt %d/%d for table %s.", i+1, attempts, tableName)
+			_, err = conn.ExecContext(ctx, `CANCEL JOBS (
+				WITH x AS (SHOW JOBS)
+				SELECT job_id FROM x WHERE job_id = $1 AND status IN ('pending', 'running', 'retrying')
+			)`, jobID)
+			if err != nil {
+				t.Fatal(err)
+			}
+		}
+
+		// Block until the job is complete.
+		var status string
+		err = conn.QueryRowContext(
+			ctx,
+			`WITH x AS (SHOW JOBS WHEN COMPLETE (SELECT $1)) SELECT status FROM x`,
+			jobID,
+		).Scan(&status)
+		if err != nil {
+			t.Fatal(err)
+		}
+		t.L().PrintfCtx(ctx, "Import job for attempt %d/%d for table %s (%d files) completed with status %s.",
+			i+1, attempts, tableName, len(files), status)
+
+		if status == "succeeded" {
+			// Remove the files that succeeded so we don't try to import
```

Comment:
Is the intention that we remove from filesToImport after the IMPORT _succeeds_, i.e. after the last attempt? If so, I don't think I understand the purpose of this codeblock, as this function will return after the last import attempt finishes. (note that after the import cancels, the job's status will be `cancelled`)